spec/java/util/regex/Pattern_n_Matcher.lsl

```diff
@@ -0,0 +1,432 @@
+    {
+        action TODO();
+    }
+
+
+    @ParameterizedResult("java.util.regex.MatchResult")
+    fun results (@target self: Matcher): Stream
+    {
+        action TODO();
+    }
+
+
+    fun start (@target self: Matcher): int
+    {
+        action TODO();
+    }
+
+
+    fun start (@target self: Matcher, arg0: String): int
+    {
+        action TODO();
+    }
+
+
+    fun start (@target self: Matcher, arg0: int): int
+    {
+        action TODO();
+    }
+
+
+    fun toMatchResult (@target self: Matcher): MatchResult
+    {
+        action TODO();
+    }
+
+
+    fun toString (@target self: Matcher): String
+    {
+        action TODO();
+    }
+
+
+    fun useAnchoringBounds (@target self: Matcher, arg0: boolean): Matcher
+    {
+        action TODO();
+    }
+
+
+    fun usePattern (@target self: Matcher, arg0: Pattern): Matcher
+    {
+        action TODO();
+    }
+
+
+    fun useTransparentBounds (@target self: Matcher, arg0: boolean): Matcher
+    {
+        action TODO();
+    }
+
+}
+
+
+
+
+automaton PatternAutomaton
+: Pattern
+{
+    // states and shifts
+
+    initstate Initialized;
+
+
+    // constructors
+
+    @private constructor Pattern (@target self: Pattern, arg0: String, arg1: int)
+    {
+        action TODO();
+    }
+
+
+    // utilities
+
+    // static methods
+
+    @static fun compile (arg0: String): Pattern
+    {
+        action TODO();
+    }
+
+
+    @static fun compile (arg0: String, arg1: int): Pattern
+    {
+        action TODO();
+    }
+
+
+    @static fun matches (arg0: String, arg1: CharSequence): boolean
+    {
+        action TODO();
+    }
+
+
+    @static fun quote (arg0: String): String
+    {
+        action TODO();
+    }
+
+
+    // methods
+
+    @ParameterizedResult("java.lang.String")
+    fun asMatchPredicate (@target self: Pattern): Predicate
+    {
+        // #problem: a lambda object should be returned here
+        // #solution: a new type with automaton with a single method
+        action NOT_IMPLEMENTED();
+    }
+
+
+    @ParameterizedResult("java.lang.String")
+    fun asPredicate (@target self: Pattern): Predicate
+    {
+        // #problem: a lambda object should be returned here
+        // #solution: a new type with automaton with a single method
+        action NOT_IMPLEMENTED();
+    }
+
+
+    fun flags (@target self: Pattern): int
+    {
+        action TODO();
+    }
+
+
+    fun matcher (@target self: Pattern, arg0: CharSequence): Matcher
+    {
+        action TODO();
+    }
+
+
+    fun pattern (@target self: Pattern): String
+    {
+        action TODO();
+    }
+
+
+    fun split (@target self: Pattern, arg0: CharSequence): array<String>
+    {
+        action TODO();
+    }
+
+
+    fun split (@target self: Pattern, arg0: CharSequence, arg1: int): array<String>
+    {
+        action TODO();
+    }
+
+
+    @ParameterizedResult("java.lang.String")
+    fun splitAsStream (@target self: Pattern, arg0: CharSequence): Stream
```

Comment:
Do we need add @final annotation to CharSequence arg0 ?

Reply:
Fixed